File: src/main/scala/org/zezutom/util/scala/InputHandler.scala
package org.zezutom.util.scala

import scala.collection.mutable.ArrayBuffer
import scala.io.Source

/**
 * Handles console / file input
 */
class InputHandler {

  def readNumbers(): Array[Int] = {
    val nums = ArrayBuffer[Int]()

    for (line <- io.Source.stdin.getLines) {
      nums += line.toInt
    }
    nums.toArray
  }

  def readNumbers(path: String): Array[Int] = {
    val nums = ArrayBuffer[Int]()

    for (line <- getLines(path)) {
      nums += line.toInt
    }
    nums.toArray
  }

  def readGraph(path: String): Map[Int, Array[Int]] = {
    var graph: Map[Int, Array[Int]] = Map()

    for (line <- getLines(path)) {
      val nums = line.split("[^0-9]+")
      graph += (nums.head.toInt -> nums.tail.map(_.toInt))
    }
    graph
  }

  private def getLines(path: String): Iterator[String] = {
    Source.fromFile(path).getLines()
  }
}

File: src/main/java/org/zezutom/util/java/InputHandler.java
package org.zezutom.util.java;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

/**
 * Handles console / file input
 */
public class InputHandler {

    public int[] readFromFile(String path) {
        List<Integer> numList = new ArrayList<>();
        try(BufferedReader br = new BufferedReader(new FileReader(path))) {
            while (br.ready()) {
                numList.add(Integer.valueOf(br.readLine()));
            }
        } catch (IOException e) {
            throw new RuntimeException("Couldn't read from " + path);
        }

        int[] nums = new int[numList.size()];

        for (int i = 0; i < numList.size(); i++) {
            nums[i] = numList.get(i);
        }
        return nums;
    }
}
